python/lsst/ctrl/bps/drivers.py

```diff
@@ -302,14 +319,21 @@ def submit_driver(config_file, **kwargs):
     """
     kwargs.setdefault("runWmsSubmissionChecks", True)
 
+    _LOG.info("DISCLAIMER: All values regarding memory consumption reported below are approximate and may "
+              "not accurately reflect actual memory usage by the bps process.")
+
     _LOG.info("Starting submission process")
-    with time_this(log=_LOG, level=logging.INFO, prefix=None, msg="Completed entire submission process"):
+    with time_this(log=_LOG, level=logging.INFO, prefix=None, msg="Completed entire submission process",
+                   mem_usage=True, mem_units="GB"):
         wms_workflow_config, wms_workflow = prepare_driver(config_file, **kwargs)
 
         _LOG.info("Starting submit stage")
-        with time_this(log=_LOG, level=logging.INFO, prefix=None, msg="Completed submit stage"):
+        with time_this(log=_LOG, level=logging.INFO, prefix=None, msg="Completed submit stage",
+                       mem_usage=True, mem_units="GB"):
             submit(wms_workflow_config, wms_workflow)
             _LOG.info("Run '%s' submitted for execution with id '%s'", wms_workflow.name, wms_workflow.run_id)
+    _LOG.info("Peak memory usage for bps process %.3f GB (main), %.3f GB (largest child process)",
+              *get_mem_peaks())
```

Comment:
You could pass in the desired output unit to `get_mem_peaks()` although I'd need to look up how we control `Quantity` stringification since `%.3f` is not going to work.

Reply:
Just for the record, after modifying **lsst.utils.timer** `get_peak_mem_usage()` (previously `get_mem_peaks()`) returns now a tuple of `astropy.units.Quantities` which are stringified with f-strings. To avoid unnecessary evaluations of these f-strings these log messages are guarded by `_LOG.isEnabledFor(logging.INFO)`.